### PR TITLE
Fix for Session Key close unexpectedly (7.8) (#5287)

### PR DESCRIPTION
**Pull request by tnleeuw**

### CODE REVIEW

**Review by jkosternl — APPROVED** [review]
I didn't read everything, since I expect it is the same code as into master.

**Comment by tnleeuw** [comment]
> I didn't read everything, since I expect it is the same code as into master.

It is virtually the same code as in master indeed, although there are some minor differences to account for differences between 7.8 and master code bases.

I put the PR back into "DRAFT" however because in master the issue is still not entirely solved and I believe I now know what the remaining problem is, and we might as well fix it fully in 7.8 in a single PR then.

**Comment by tnleeuw** [comment]
Oops, I forgot to run IAF-Test before marking the PR as "Ready for Review" again and it turns out we have a number of failures in there still that I need to investigate now.

**Comment by tnleeuw** [comment]
> Oops, I forgot to run IAF-Test before marking the PR as "Ready for Review" again and it turns out we have a number of failures in there still that I need to investigate now.

I checked the results in Jenkins and it appears to be just Websphere flakiness, and locally on my laptop it appears to be a problem with Docker / ActiveMQ? Not able to connect to ActiveMQ running in the Docker image, which causes a number of tests to fail.

So the failing tests do not appear related to my actual changes.

File: core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderStub.java [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
-
+			session.getCloseables().clear();
+		}
 		session = pipelineSession;
+		for (String key : session.keySet()) {
+			if (!(session.get(key) instanceof InputStream)) {
+				// Do not copy input-streams b/c then they are no longer readable
+				sessionCopy.put(key, session.getMessage(key).copyMessage());
+			}
+		}
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderStub.java:31]: Ik snap dit niet helemaal maar het is een (oude) test..

**Review by nielsm5 — APPROVED** [review]

File: core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderStub.java (hunk 1 above)
Comment (tnleeuw) [reply core/src/test/java/nl/nn/adapterframework/http/cxf/SoapProviderStub.java:31]: > Ik snap dit niet helemaal maar het is een (oude) test..

De test probeert messages te lezen die in de session zitten, maar omdat de session gesloten is zijn die messages nu NULL.
Dus kopieer ik de session inhoud en maak een kopie van iedere session value.

Alleen niet als het een input stream is want dan lees je die stream en ben je hem kwijt. (Er is 1 test die een input stream in de session doet).

Ik had misschien ook alleen de instance-of-message values kunnen kopiëren.

**Pull request merged**